registry/storage/catalog.go

```diff
@@ -40,22 +41,20 @@ func (reg *registry) Repositories(ctx context.Context, repos []string, last stri
 
 		// if we've filled our array, no need to walk any further
 		if len(foundRepos) == len(repos) {
-			return errFinishedWalk
+			finishedWalk = true
```

Comment:
Can you also remove the `errFinishedWalk` variable

Reply:
Done.